File: TheAlchemy/src/main/java/com/qualityplus/alchemist/listener/InventoryListener.java
package com.qualityplus.alchemist.listener;

import com.qualityplus.alchemist.base.gui.AlchemistGUI;
import eu.okaeri.platform.core.annotation.Component;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.inventory.InventoryClickEvent;
import org.bukkit.event.inventory.InventoryCloseEvent;
import org.bukkit.event.inventory.InventoryDragEvent;
import org.bukkit.event.inventory.InventoryOpenEvent;

/**
 * Inventory Listener
 */
@Component
public final class InventoryListener implements Listener {
    /**
     *
     * @param event {@link InventoryClickEvent}
     */
    @EventHandler
    public void onInventoryClick(final InventoryClickEvent event) {
        if (event.getClickedInventory() != null && event.getInventory().getHolder() != null) {
            if (event.getInventory().getHolder() instanceof AlchemistGUI) {
                ((AlchemistGUI) event.getInventory().getHolder()).onInventoryClick(event);
            }
        }
    }

    /**
     *
     * @param event {@link InventoryCloseEvent}
     */
    @EventHandler
    public void onInventoryClose(final InventoryCloseEvent event) {
        if (event.getInventory().getHolder() != null) {
            if (event.getInventory().getHolder() instanceof AlchemistGUI) {
                ((AlchemistGUI) event.getInventory().getHolder()).onInventoryClose(event);
            }
        }
    }

    /**
     *
     * @param event {@link InventoryOpenEvent}
     */
    @EventHandler
    public void onInventoryOpen(final InventoryOpenEvent event) {
        if (event.getInventory().getHolder() != null) {
            if (event.getInventory().getHolder() instanceof AlchemistGUI) {
                ((AlchemistGUI) event.getInventory().getHolder()).onInventoryOpen(event);
            }
        }
    }

    /**
     *
     * @param event {@link InventoryDragEvent}
     */
    @EventHandler
    public void onInventoryOpen(final InventoryDragEvent event) {
        if (event.getInventory().getHolder() != null) {
            if (event.getInventory().getHolder() instanceof AlchemistGUI) {
                ((AlchemistGUI) event.getInventory().getHolder()).onInventoryDrag(event);
            }
        }
    }
}

File: TheAnvil/src/main/java/com/qualityplus/anvil/listener/InventoryListener.kt
package com.qualityplus.anvil.listener

import com.qualityplus.anvil.base.gui.AnvilGUI
import eu.okaeri.platform.core.annotation.Component
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.inventory.InventoryClickEvent
import org.bukkit.event.inventory.InventoryCloseEvent
import org.bukkit.event.inventory.InventoryDragEvent
import org.bukkit.event.inventory.InventoryOpenEvent

@Component
class InventoryListener : Listener {
    @EventHandler
    fun onInventoryClick(event: InventoryClickEvent) {
        if (event.clickedInventory != null && event.inventory.holder != null) if (event.inventory.holder is AnvilGUI) (event.inventory.holder as AnvilGUI?)!!.onInventoryClick(
            event
        )
    }

    @EventHandler
    fun onInventoryClose(event: InventoryCloseEvent) {
        if (event.inventory.holder != null) if (event.inventory.holder is AnvilGUI) (event.inventory.holder as AnvilGUI?)!!.onInventoryClose(
            event
        )
    }

    @EventHandler
    fun onInventoryOpen(event: InventoryOpenEvent) {
        if (event.inventory.holder != null) if (event.inventory.holder is AnvilGUI) (event.inventory.holder as AnvilGUI?)!!.onInventoryOpen(
            event
        )
    }

    @EventHandler
    fun onInventoryOpen(event: InventoryDragEvent) {
        if (event.inventory.holder != null) if (event.inventory.holder is AnvilGUI) (event.inventory.holder as AnvilGUI?)!!.onInventoryDrag(
            event
        )
    }
}